src/soc/hw/noc/test/test_noc_buffer.py

```diff
@@ -0,0 +1,387 @@
+"""
+Copyright (c) 2019 by the author(s)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+=============================================================================
+
+Test for the module 'noc_buffer'
+
+Author(s):
+ - Max Koenen <max.koenen@tum.de>
+"""
+
+import random
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import RisingEdge, FallingEdge
+from cocotb.result import TestFailure
+
+# Mirroring expected contents of the FIFO
+_buffer_flits = []
+_buffer_last = []
+
+@cocotb.test()
+def test_noc_buffer_fullpacket(dut):
+    """
+    Test the module noc_buffer with the fullpacket feature enabled.
+    This test is hardcoded and only works with WIDTH = 8, DEPHT = 4 and
+    FULLPACKET = 1.
+    """
+
+    if (dut.FULLPACKET.value.integer != 1 or dut.FLIT_WIDTH.value.integer != 8 or
```

Comment:
nit: You don't need the `(` in Python.

Reply:
I do when I have a line break, no? (at least it doesn't work for me without). I could put it all in the same line of course, was just trying to avoid too long lines.

Reply:
Ah that's true, ignore my comment then. Sorry for the noise.